pulsar-jms/src/test/java/com/datastax/oss/pulsar/jms/TemporaryDestinationsNonAdminTest.java

```diff
@@ -0,0 +1,100 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.pulsar.jms;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import com.datastax.oss.pulsar.jms.utils.PulsarContainerExtension;
+import java.util.Map;
+import javax.jms.Connection;
+import javax.jms.Destination;
+import javax.jms.IllegalStateException;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@Slf4j
+public class TemporaryDestinationsNonAdminTest {
+
+    @RegisterExtension
+    static PulsarContainerExtension pulsarContainer =
+            new PulsarContainerExtension()
+                    .withEnv("PULSAR_PREFIX_allowAutoTopicCreation", "true")
+                    .withEnv("PULSAR_PREFIX_allowAutoTopicCreationType", "non-partitioned")
+                    .withEnv("PULSAR_PREFIX_transactionCoordinatorEnabled", "false");
+
+    @Test
+    public void allowTemporaryTopicWithoutAdminTest() throws Exception {
+        Map<String, Object> properties = getJmsProperties();
+        properties.put("jms.allowTemporaryTopicWithoutAdmin", "true");
+        useTemporaryDestinationNonAdminTest(properties, false);
+    }
+
+    @Test
+    public void forbidTemporaryTopicWithoutAdminTest() throws Exception {
+        Map<String, Object> properties = getJmsProperties();
+        useTemporaryDestinationNonAdminTest(properties, true);
+    }
+
+    @NotNull
+    private static Map<String, Object> getJmsProperties() {
+        Map<String, Object> properties = pulsarContainer.buildJMSConnectionProperties();
+        properties.put("jms.forceDeleteTemporaryDestinations", "true");
+        properties.put("jms.usePulsarAdmin", "false");
+        return properties;
+    }
+
+    private void useTemporaryDestinationNonAdminTest(Map<String, Object> properties, boolean expectAdminErrors)
+            throws Exception {
+
+        try (PulsarConnectionFactory factory = new PulsarConnectionFactory(properties)) {
+            try (Connection connection = factory.createConnection()) {
+                connection.start();
+                try (Session session = connection.createSession()) {
+                    if (expectAdminErrors) {
+                        assertThrows(IllegalStateException.class, session::createTemporaryTopic);
```

Comment:
@eolivelli 
- `IllegalStateException` is a checked Exception extending `JMSException` in JMS 2.x. 

- `Utils.handleException` is casting `IllegalStateException` into `JMSException` and returning.  assertThrows passes because it uses isInstance() method of Class.

- I am making it `JMSException` in the test for clarity,

Reply:
oh, thanks I thought it was java.lang.IllegalStateException